Integrations/SplunkPy/SplunkPy.py

```diff
@@ -262,82 +262,117 @@ def request(url, message, **kwargs):
     headers = ""
-    if res and len(res) > 0:
-        if not isinstance(res[0], dict):
+    if parsed_search_results and len(parsed_search_results) > 0:
+        if not isinstance(parsed_search_results[0], dict):
             headers = "results"
 
-    human_readable = tableToMarkdown("Splunk Search results \n\n Results for query: {}".format(demisto.args()['query']),
-                                     res, headers)
+    human_readable = tableToMarkdown("Splunk Search results \n\n Results for query: {}".format(args['query']),
+                                     parsed_search_results, headers)
+    return human_readable
+
+
+def get_current_results_batch(search_job, batch_size, results_offset):
+    current_batch_kwargs = {
+        "count": batch_size,
+        "offset": results_offset
+    }
+
+    results_batch = search_job.results(**current_batch_kwargs)
+    return results_batch
+
+
+def parse_batch_of_results(current_batch_of_results, max_results_to_add):
+    parsed_batch_results = []
+    batch_dbot_scores = []
+    results_reader = results.ResultsReader(io.BufferedReader(ResponseReaderWrapper(current_batch_of_results)))
+    for item in results_reader:
+        if isinstance(item, results.Message):
+            if "Error in" in item.message:
+                raise ValueError(item.message)
+            parsed_batch_results.append(convert_to_str(item.message))
+
+        elif isinstance(item, dict):
+            if demisto.get(item, 'host'):
+                batch_dbot_scores.append({'Indicator': item['host'], 'Type': 'hostname',
+                                          'Vendor': 'Splunk', 'Score': 0, 'isTypedIndicator': True})
+            # Normal events are returned as dicts
+            parsed_batch_results.append(item)
+
+        if len(parsed_batch_results) >= max_results_to_add:
+            break
+    return parsed_batch_results, batch_dbot_scores
+
+
+def splunk_search_command():
+    args = demisto.args()
+
+    query = build_search_query(args)
+    search_kwargs = build_search_kwargs(args)
+    search_job = service.jobs.create(query, **search_kwargs)  # type: ignore
+    num_of_results = search_job["resultCount"]
+
+    results_limit = int(demisto.args().get("event_limit", 100))
+    if results_limit == 0:
+        # In Splunk, a result limit of 0 means no limit.
+        results_limit = float("inf")
+    batch_size = int(demisto.args().get("batch_limit", 25000))
+
+    results_offset = 0
+    parsed_search_results = []  # type: List[Dict[str,Any]]
+    dbot_scores = []  # type: List[Dict[str,Any]]
+
+    while len(parsed_search_results) < int(num_of_results) and len(parsed_search_results) < results_limit:
+        current_batch_of_results = get_current_results_batch(search_job, batch_size, results_offset)
+        max_results_to_add = results_limit - len(parsed_search_results)
+        parsed_batch_results, batch_dbot_scores = parse_batch_of_results(current_batch_of_results, max_results_to_add)
```

Comment:
Not sure that this is clear enough

Reply:
I agree, I think it's fairly clear but not very clear, 6/10 maybe. Needs improvement.
I can change the naming to raw_results_batch and results_batch, but I could not find any other way to improve it (thought about it a lot - the batching mechanism and the need for efficiency are damaging readability inherently IMHO).
Would love to hear ideas for improvement

Reply:
Lets do this today together, this is not clear enough for me

Reply:
Done